lib/paf.ml

```diff
@@ -337,8 +339,11 @@ let serve_when_ready :
      let rec loop () =
        accept t >>= function
        | Ok flow ->
-           Lwt.async (fun () -> handler flow) ;
-           Lwt.pause () >>= loop
+        Lwt.async (fun () -> (connect flow >>= function
+        | Ok flow -> handler flow
+        | Error `Closed -> Logs.info (fun m -> m "Error: closed"); Lwt.return ()
```

Comment:
I misunderstood the meaning of `'Closed` here and it should probably only be raised by the `accept` function, so in the outer loop